test/libsolidity/solidityExecutionFramework.h

```diff
@@ -284,45 +264,90 @@ class ExecutionFramework
 protected:
 	void sendMessage(bytes const& _data, bool _isCreation, u256 const& _value = 0)
 	{
-		m_state.addBalance(m_sender, _value); // just in case
-		eth::Executive executive(m_state, m_envInfo, m_sealEngine.get());
-		eth::ExecutionResult res;
-		executive.setResultRecipient(res);
-		eth::Transaction t =
-			_isCreation ?
-				eth::Transaction(_value, m_gasPrice, m_gas, _data, 0, KeyPair::create().sec()) :
-				eth::Transaction(_value, m_gasPrice, m_gas, m_contractAddress, _data, 0, KeyPair::create().sec());
-		bytes transactionRLP = t.rlp();
-		try
-		{
-			// this will throw since the transaction is invalid, but it should nevertheless store the transaction
-			executive.initialize(&transactionRLP);
-			executive.execute();
-		}
-		catch (...) {}
+		RPCRequest::transactionData d;
+		d.data = "0x" + toHex(_data);
+		d.from = m_testAccount;
+		d.gas = toHex(m_gas, HexPrefix::Add);
+		d.gasPrice = toHex(m_gasPrice, HexPrefix::Add);
+		d.value = toHex(_value, HexPrefix::Add);
+		if (_isCreation)
+			d.to = "";
+		else
+			d.to = dev::toString(m_contractAddress);
+
+
+		string hash = m_socket.eth_sendTransaction(d);
+		m_socket.test_mineBlocks(1);
+		RPCRequest::transactionReceipt receipt;
+		receipt = m_socket.eth_getTransactionReceipt(hash);
+		string output = m_socket.eth_call(d, "latest");
+
 		if (_isCreation)
 		{
-			BOOST_REQUIRE(!executive.create(m_sender, _value, m_gasPrice, m_gas, &_data, m_sender));
-			m_contractAddress = executive.newAddress();
+			m_contractAddress = Address(receipt.contractAddress);
 			BOOST_REQUIRE(m_contractAddress);
-			BOOST_REQUIRE(m_state.addressHasCode(m_contractAddress));
+			string code = m_socket.eth_getCode(receipt.contractAddress, "latest");
+			BOOST_REQUIRE(code.size() > 2);
 		}
 		else
 		{
-			BOOST_REQUIRE(m_state.addressHasCode(m_contractAddress));
-			BOOST_REQUIRE(!executive.call(m_contractAddress, m_sender, _value, m_gasPrice, &_data, m_gas));
+			string code = m_socket.eth_getCode(d.to, "latest");
```

Comment:
This is done before the actual call. That's also important because the transaction could call selfdestruct.